sympy/printing/rust.py

```diff
@@ -438,10 +438,15 @@ def _print_ITE(self, expr):
         return self._print(_piecewise)
 
     def _print_MatrixBase(self, A):
-        if A.cols == 1:
+        if A.cols == 1 or A.rows == 1:
```

Comment:
How do matrices work in rust? I would normally say that changing the dimensionality of a matrix like this is a bad idea.

Reply:
There is no built-in matrix support apart from fixed size arrays. We could import a crate, but this is a bit opiniated (ndarray/nalgebra are two common options). I've implemented with the idea the type can be substituted with the wanted structure, as most crates support this form of indexing.

Is there a difference between a row/column vector and a matrix with one dimension set to one? The handling of a one-dimensional vector is significantly easier than the matrix case above.

Reply:
Well for one you would write A[0][0], A[0][1], ..., and for the other that is an error. Does Rust distinguish between shape (n, 1) and shape (n)? If not, then what is already here could make sense.  

It also depends on how the matrix libraries work. Do they automatically treat one-dimensional vectors as a row or column vector depending on the context? And even if they do, you lose the ability to get shape checking errors.  NumPy does this, for instance, and it leads to a lot of problems (in my opinion). See my comments at https://github.com/sympy/sympy/issues/18824.

Reply:
Rust has quite strict indexing rules, where the indices usually must match the number of dimensions, so shape(n, 1) != shape(n). I am a bit fuzzy on the sympy parts of this. Is there a concept of a vector, or are they simply 1D matrices (with # of cols == 1)?

Reply:
SymPy matrices always have 2 dimensions. If Rust works like that, then I would say the code that was already there is wrong. We do have some vector objects, but they aren't part of the matrices module.

Reply:
So `_print_MatrixBase` should never be called to print a vector? In that case it is tempting to remove the entire conditional and only use the true matrix printer

Reply:
SymPy's vector objects are based on the concept of a vector in Physics and are very different from the array concept of a "vector"

Reply:
There is also Indexed and NDimArray which are supposed to represent arrays of any dimension and should be usable for codegen.